src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateInstanceEventHandler.java

```diff
@@ -0,0 +1,106 @@
+package org.folio.inventory.dataimport.handlers.actions;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.domain.instances.InstanceCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.inventory.support.InstanceUtil;
+import org.folio.processing.exceptions.EventProcessingException;
+import org.folio.processing.mapping.defaultmapper.RecordToInstanceMapperBuilder;
+import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.jaxrs.model.Record;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+public class UpdateInstanceEventHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(UpdateInstanceEventHandler.class);
+  private static final String MARC_KEY = "MARC";
+  private static final String MAPPING_RULES_KEY = "MAPPING_RULES";
+  private static final String MAPPING_PARAMS_KEY = "MAPPING_PARAMS";
+
+  private Storage storage;
+  private Context context;
+
+  public UpdateInstanceEventHandler(Storage storage, Context context) {
+    this.storage = storage;
+    this.context = context;
+  }
+
+  public CompletableFuture<Instance> handle(Map<String, String> eventPayload) {
+    CompletableFuture<Instance> future = new CompletableFuture<>();
+    try {
+      if (eventPayload == null || isEmpty(eventPayload.get(MARC_KEY)) || isEmpty(eventPayload.get(MAPPING_RULES_KEY)) || isEmpty(eventPayload.get(MAPPING_PARAMS_KEY))) {
+        String message = "Event does not contain required data to update Instance";
+        LOGGER.error(message);
+        future.completeExceptionally(new EventProcessingException(message));
+        return future;
+      }
+      JsonObject mappingRules = new JsonObject(eventPayload.get(MAPPING_RULES_KEY));
+      MappingParameters mappingParameters = new JsonObject(eventPayload.get(MAPPING_PARAMS_KEY)).mapTo(MappingParameters.class);
+      Record marcRecord = new JsonObject(eventPayload.get(MARC_KEY)).mapTo(Record.class);
+
+      JsonObject parsedRecord = JsonObject.mapFrom(marcRecord.getParsedRecord().getContent());
+      String instanceId = marcRecord.getExternalIdsHolder().getInstanceId();
```

Comment:
@KaterynaSenchenko  I think that there will be exception if ExternalIdsHolder will be empty. I think it will be better to handle it.

Reply:
This Instance update is triggered by the update of the underlying SRS MARC Record - it must contain the id of the Instance it is linked to. In case there is no Instance id (which we do not expect), the exception will be caught below, logged and the future will be completed exceptionally

Reply:
@KaterynaSenchenko Thank you for the description!